addons/hr_expense/tests/test_expenses.py

```diff
@@ -1457,3 +1458,29 @@ def test_expense_by_company_with_caba_tax(self):
         moves = expense_sheet.action_sheet_move_create()
         tax_lines = moves.line_ids.filtered(lambda line: line.tax_line_id == caba_tax)
         self.assertNotEqual(tax_lines.account_id, caba_transition_account, "The tax should not be on the transition account")
+
+    def test_expense_set_total_amount_to_0(self):
+        """Checks that amount fields are nullified when setting total_amount to 0"""
+        expense = self.env['hr.expense'].create({
+            'name': 'Expense with amount',
+            'employee_id': self.expense_employee.id,
+            'product_id': self.product_c.id,
+            'total_amount': 100.0,
+            'tax_ids': self.tax_purchase_a.ids,
+        })
+        expense.total_amount = 0.0
+        self.assertEqual(float_compare(expense.amount_tax, 0, expense.currency_id.rounding), 0)
+        self.assertEqual(float_compare(expense.total_amount_company, 0, expense.currency_id.rounding), 0)
+
+    def test_expense_set_quantity_to_0(self):
+        """Checks that amount fields except for unit_amount are nullified when setting quantity to 0"""
+        expense = self.env['hr.expense'].create({
+            'name': 'Expense with amount',
+            'employee_id': self.expense_employee.id,
+            'product_id': self.product_b.id,
+            'quantity': 10,
+            'tax_ids': self.tax_purchase_a.ids,
+        })
+        expense.quantity = 0
+        self.assertEqual(float_compare(expense.total_amount, 0, expense.currency_id.rounding), 0)
+        self.assertEqual(float_compare(expense.unit_amount, self.product_b.standard_price, expense.currency_id.rounding), 0)
```

Comment:
Same, use currency_id or company_currency_id .is_zero method
Also same as above, unit_amount is company_currency_id, not currency_id (even if it's the same, it's better to use the right one)